File: C/duracao.c
#include <stdio.h>
#include <stdlib.h>

int main() {

    int duracao, segundos, minutos, horas, rest;

    printf("Digite a duracao em segundos:");
    scanf("%d", &duracao);

    horas = duracao / 3600;
    rest = duracao % 3600;

    minutos = rest / 60;
    segundos = rest % 60;

    printf("%d:%d:%d",horas, minutos, segundos);
    return 0;
}

File: java/duracao.java
import java.util.Locale;
import java.util.Scanner;

public class duracao {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int minutos, horas, segundos;
		
		System.out.print("Digite a duracao em segundos: ");
		segundos = sc.nextInt();
		
		horas = segundos / 3600;
		minutos = (segundos % 3600) / 60;
		segundos = (segundos % 3600) % 60;
		
		System.out.println( horas + ":" +  minutos + ":" + segundos);
		
		sc.close();
	}

}
